ls1intum/Artemis
# [Refactoring] JenkinsService refactoring

### Checklist

### Motivation and Context


The `JenkinsService` class contains code which can be extracted in dedicated utility classes in order to reduce complexity.

### Description

In this PR we extract code from the `JenkinsService` class. We move all build plan related related code into the `BuildPlanService` class. We also create a `JenkinsBuildPlanUtils.java` which contains utility methods used by the build plan service.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Generate a Programming Exercise
4. Participate in the programming exercise and make sure that you get the correct results.
